src/compat/dbtype.h

```diff
@@ -772,6 +772,7 @@ union db_char
   {
     unsigned char style;
     unsigned char codeset;
+    bool is_max_string;
```

Comment:
I have added `is_max_string` to `compat/dbi_compat.h`,  and also changed that handling over the `else`. Hope it looks better.